### DISCUSSION TITLE
docs: Consider using sqlite over postgres in getting started guides

### DISCUSSION THREAD

**Discussion by pnadolny13** [meltano/meltano#6950] [General]
Related to https://github.com/meltano/internal-data/issues/39#issuecomment-1297715601 (private) and https://github.com/meltano/internal-data/issues/38#issuecomment-1297723663 (private).

When digging into what new users are doing we see high usage of tap and target postgres. As part of that we see a fair amount of failures related to target postgres. I also know that we have a troubleshooting section in the hub for target-postgres since we hear a lot of the same issues with target-postgres (e.g. [psycopg2 install error](https://stackoverflow.com/questions/26288042/error-installing-psycopg2-library-not-found-for-lssl)).

I was wondering if we might want to consider switching to something like target-sqlite for the tutorials. When I tested that target you literally dont need to configure anything and it's almost guaranteed that if a new user can run meltano then they can use sqlite (since its the default system db). On the other hand target-postgres requires a docker container or a pre-configured local postgres and other settings to be configured in meltano.

What do you think? cc @tayloramurphy @DouweM @sbalnojan

**Comment by sbalnojan** [comment]
@pnadolny13 Maybe, to get people to first run that will help. 

But I'm a bit torn on this subject as I think it's of high importance to provide a great tap & target postgres. So instead (or decoupled from this) we could prioritize work on the high quality taps & targets that just work. 

I think it's always a fine line in a tutorial to balance not loosing people vs. making it too hard. And I'm not sure where this one is.

**Reply by aaronsteers** [reply]
@pnadolny13 and @sbalnojan - I think SQLite has some limitations we've run into recently.

DuckDB is getting broad support though and is pip installable - what about moving all our internal examples from postgres to duckdb? 

Because it is local, self-contained, and pip installable, I think the chance of misconfiguration is extremely low.

Maybe out DuckDB integrations are not mature enough support today, but if we fast forward 3 weeks or so, I think the maturity and stability of those plugins will likely be there in a reasonable enough timeframe that we could switch over on/after Thanksgiving week.

MDS in a box already uses this already:

- https://github.com/matsonj/nba-monte-carlo/blob/master/meltano.yml#L53-L57

And there's no pre-initialization needed as far as I'm aware.

**Reply by tayloramurphy** [reply]
@sbalnojan I'm a plus one on making the getting started guide as simple as possible. We want people to get to the full experience and aha moment as quick as possible with as few chances to trip them up as possible.

The really nice thing about duckdb is that there's basically zero configuration that people would need to do. It "just works" with minimal creds _and_ dbt would work with it too. 

If we could use a data source that doesn't require auth (some listed here https://mixedanalytics.com/blog/list-actually-free-open-no-auth-needed-apis/) then the setup experience would be smoother and minimal. That said, github auth isn't _too_ bad. 

> we could prioritize work on the high quality taps & targets that just work

We're definitely already doing this with postgres and snowflake, but I'd rather simplify the getting started guide as much as possible separate from that work.

**Reply by aaronsteers** [reply]
Agreed with the above from @tayloramurphy. Another nice thing about DuckDB over SQLite is that we can be really explicit about the pip-installed DuckDB version and avoid the version issues we sometimes see with SQLite: #6405

**Comment by sbalnojan** [comment]
I'd like to pick up @tayloramurphy idea and make the jump off of github as well. Related to https://github.com/meltano/meltano/pull/6984.
